datahub-web-react/src/app/entity/shared/components/styled/DeprecationPill.tsx

```diff
@@ -89,12 +114,33 @@ export const DeprecationPill = ({ deprecation, preview }: Props) => {
                         {isDividerNeeded && <ThinDivider />}
                         {deprecation?.note !== '' && <DeprecatedSubTitle>{deprecation.note}</DeprecatedSubTitle>}
                         {deprecation?.decommissionTime !== null && (
-                            <Typography.Text type="secondary">
-                                <Tooltip placement="right" title={decommissionTimeGMT}>
-                                    <LastEvaluatedAtLabel>{decommissionTimeLocal}</LastEvaluatedAtLabel>
-                                </Tooltip>
-                            </Typography.Text>
+                            <>
+                                <Typography.Text type="secondary">
+                                    <Tooltip placement="right" title={decommissionTimeGMT}>
+                                        <LastEvaluatedAtLabel>{decommissionTimeLocal}</LastEvaluatedAtLabel>
+                                    </Tooltip>
+                                </Typography.Text>
+                            </>
                         )}
+                        {isDividerNeeded && <ThinDivider />}
+                        <Button
+                            type="default"
+                            onClick={() =>
+                                Modal.confirm({
+                                    title: `Confirm Mark as undeprecated`,
+                                    content: `Are you sure you want to mark these assets as undeprecated?`,
+                                    onOk() {
+                                        batchUndeprecate();
+                                    },
+                                    onCancel() {},
+                                    okText: 'Yes',
+                                    maskClosable: true,
+                                    closable: true,
+                                })
+                            }
+                        >
+                            Mark as un-deprecated
```

Comment:
I also don't think we want this button to be shown every time we show this pill. ie. we don't want them to be able to undeprecate from the search results page so `DefaultPrevieCard` instances of this pill shouldn't show this button.

What do you think about adding a new optional prop called `showUndeprecate` that will show this button if you pass it in?

Reply:
Checked and Fixed!!